doc/source/building/index.rst

```diff
@@ -0,0 +1,354 @@
+        --version`` or ``ifort --version``).
+
+
+Building SciPy from source
+--------------------------
+
+If you want to only install SciPy from source once and not do any development
+work, then the recommended way to build and install is to use ``pip``:
+
+.. tab-set::
+
+  .. tab-item:: Virtual env or system Python
+
+    ::
+
+      # To build the latest stable release:
+      pip install scipy --no-binary scipy
+
+      # To build a development version, you need a local clone of the SciPy git repository:
+      git clone https://github.com/scipy/scipy.git
+      git submodule update --init
+      pip install .
+
+  .. tab-item:: Conda env
+
+    If you are using a conda environment, ``pip`` is still the tool you use to
+    invoke a from-source build of SciPy. It is important to always use the
+    ``--no-build-isolation`` flag to the ``pip install`` command, to avoid
+    building against a ``numpy`` wheel from PyPI. In order for that to work you
+    must first install the remaining build dependencies into the conda
+    environment::
+
+      # Either install all SciPy dev dependencies into a fresh conda environment
+      mamba env create -f environment.yml
+
+      # Or, install only the required build dependencies
+      mamba install python numpy cython pythran pybind11 compilers openblas pkg-config
+
+      # To build the latest stable release:
+      pip install scipy --no-build-isolation --no-binary scipy
+
+      # To build a development version, you need a local clone of the SciPy git repository:
+      git clone https://github.com/scipy/scipy.git
+      git submodule update --init
+      pip install . --no-build-isolation
+
+
+.. _the-dev-py-interface:
+
+Building from source for SciPy development
+``````````````````````````````````````````
+
+If you want to build from source in order to work on SciPy itself, first clone
+the SciPy repository::
+
+      git clone https://github.com/scipy/scipy.git
+      git submodule update --init
+
+Then you want to do the following:
+
+1. Create a dedicated development environment (virtual environment or conda
+   environment),
+2. Install all needed dependencies (*build*, and also *test*, *doc* and
+   *optional* dependencies), 
+3. Build SciPy with our ``dev.py`` developer interface.
+
+Step (3) is always the same, steps (1) and (2) are different between conda and
+virtual environments:
+
+.. tab-set::
+
+  .. tab-item:: Virtual env or system Python
+
+    .. note::
+
+       There are many tools to manage virtual environments, like ``venv``,
+       ``virtualenv``/``virtualenvwrapper, ``pyenv``/``pyenv-virtualenv``,
+       Poetry, PDM, Hatch, and more. Here we use the basic ``venv`` tool that
+       is part of the Python stdlib. You can use any other tool; all we need is
+       an activated Python environment.
+
+    Create and activate a virtual environment in a new directory named ``venv``::
+
+       python -m venv venv
+       source venv/bin/activate
```

Comment:
`source` is not a command on all Windows shells. I would find it helpful to note the equivalent command when `source` is not available. or maybe link to [How venvs work](https://docs.python.org/3/library/venv.html#how-venvs-work), which has a helpful table.

Reply:
I like linking to the table, since this is a bit of a mess that I don't want to try to maintain in our own docs.